modules/m_sasl.c

```diff
@@ -223,17 +224,30 @@ me_sasl(struct MsgBuf *msgbuf_p, struct Client *client_p, struct Client *source_
 		rb_strlcpy(target_p->localClient->sasl_agent, parv[1], IDLEN);
 
 	if(*parv[3] == 'C')
+	{
 		sendto_one(target_p, "AUTHENTICATE %s", parv[4]);
+		target_p->localClient->sasl_messages++;
+	}
 	else if(*parv[3] == 'D')
 	{
-		if(*parv[4] == 'F')
+		if(*parv[4] == 'F') {
 			sendto_one(target_p, form_str(ERR_SASLFAIL), me.name, EmptyString(target_p->name) ? "*" : target_p->name);
+			if (target_p->localClient->sasl_messages > 0)
```

Comment:
To exclude failures caused by unknown SASL mechanism (which can legitimately happen with clients lacking RPL_MECHLIST support).